classes/test/engines/concurrent.php

```diff
@@ -132,9 +132,35 @@ public function run(atoum\test $test)
 				$phpCode .= '$test->getMockGenerator()->disallowUndefinedMethodUsage();';
 			}
 
-			foreach ($test->getExtensions() as $extension)
+			$extensions = $test->getExtensions();
+
+			foreach ($extensions as $extension)
 			{
-				$phpCode .= '$test->addExtension(new ' . get_class($extension) . ');';
+				$configuration = $exception = null;
+				$extensionClass = get_class($extension);
+
+				try
+				{
+					$configuration = $extensions[$extension];
+				}
+				catch (\unexpectedValueException $exception) {}
+
+				if ($exception !== null || $configuration === null)
+				{
+					$phpCode .= '$test->addExtension(new ' . $extensionClass . ');';
+				}
+				else
+				{
+					$configurationClass = get_class($configuration);
+					$serialized = $configuration->serialize();
+
+					if (is_array($serialized) === false)
+					{
+						throw new exceptions\logic('Extension (' . $extensionClass . ') configuration (' . $configurationClass . ') should serialize as an array');
+					}
+
+					$phpCode .= '$test->addExtension(new ' . $extensionClass . ', ' . $configurationClass . '::unserialize(' . var_export($serialized, true) . '));';
```

Comment:
the first tests I made were by passing the clientId and clientToken strings to the configuration. It worked.

but, it may be better to pass the ClientConfiguration object (here, we call an undefined  __set_state method on the ClientConfigration).

what about using serialize/unserialize to do that ? 

By doing this change, passing the ClientConfiguration object to the atoum configuration works : 
```php
$phpCode .= '$test->addExtension(new ' . $extensionClass . ', ' . $configurationClass . '::unserialize(unserialize(' . var_export(serialize($serialized), true) . ')));';
```

@jubianchi what's your thoughts about this ?